File: backend/components/metiers/threed/security/rbac/engine/rbac_engine.js
// rbac_engine.js – Moteur RBAC

function checkPermission(userRole, action) {
  if (userRole === 'admin') return true;
  if (userRole === 'auditor' && action === 'audit') return true;
  if (userRole === 'manager' && ['read', 'write'].includes(action)) return true;
  if (userRole === 'user' && action === 'read') return true;
  return false;
}

module.exports = { checkPermission };

File: backend/components/metiers/threed/security/rbac/engine/rbac_engine.py
# rbac_engine.py – Moteur RBAC

def check_permission(user_role, action):
    # Exemple de logique RBAC simple
    if user_role == 'admin':
        return True
    if user_role == 'auditor' and action == 'audit':
        return True
    if user_role == 'manager' and action in ['read', 'write']:
        return True
    if user_role == 'user' and action == 'read':
        return True
    return False
